hail/python/hail/__init__.py

```diff
@@ -11,6 +11,14 @@
 del pkg_resources
 del sys
 
+
+try:
+    from IPython.core.getipython import get_ipython
+    IS_NOTEBOOK = get_ipython().__class__.__name__ == 'ZMQInteractiveShell'
+except (NameError, ModuleNotFoundError):
+    IS_NOTEBOOK = False
```

Comment:
@iris-garden  Put this in hailtop b/c hailtop isn't allowed to depend on hail (circularity!)

Reply:
moved!